psi4/src/psi4/libmints/writer.cc

```diff
@@ -333,9 +348,17 @@ void FCHKWriter::write(const std::string &filename) {
     bool pHF = (refwf != NULL);
 
     if (pHF) {
-        // Level of theory may be correlated
-        Da_ao = refwf->Da_subset("AO");
-        Db_ao = refwf->Db_subset("AO");
+        // regenerate SCF densities from coefficients matrix
+        // since one currently cannot be sure Da_subset("AO")
```

Comment:
Wait, what? The reference wavefunction's density matrices aren't actually the reference densities? If you can give me an example, I can go ahead and fix this for DCT. No guarantees about my ability to fix this for `ccenergy`, but I can at least give a look.

Reply:
For DCT i think `refwfn.D` is correct (=SCF), but for `ccenergy` it is the CC density.

Reply:
Then why is it that for DCT, we [send in `recompute_` true](https://github.com/psi4/psi4/pull/1884/files#diff-c4d6e39b37f0195e3587302260658d00R2039)? The `else` clause should do what we want for DCT in that case, right?

Reply:
hmm, oversight from me. It's no longer needed. I checked and setting it to `False` gives the same fchk file.